rpcserver.go

```diff
@@ -2995,7 +2993,7 @@ func handleGetBlockTemplateProposal(s *rpcServer, request *dcrjson.TemplateReque
 	block := dcrutil.NewBlock(&msgBlock)
 
 	// Ensure the block is building from the expected previous block.
-	expectedPrevHash, _ := s.server.blockManager.chainState.Best()
+	expectedPrevHash := &s.server.blockManager.chain.BestSnapshot().PrevHash
```

Comment:
This is incorrect.  The block needs to build on the current best chain tip, not the one before it.